--- a/third_party/blink/renderer/core/dom/document.cc
+++ b/third_party/blink/renderer/core/dom/document.cc
@@ -1045,11 +1045,6 @@
       data_(MakeGarbageCollected<DocumentData>(GetExecutionContext())) {
   TRACE_EVENT("blink", "Document::Document", perfetto::Flow::FromPointer(this));
   DCHECK(agent_);
-  if (RuntimeEnabledFeatures::FingerprintingClientRectsNoiseEnabled()) {
-    // Precompute -0.0003% to 0.0003% noise factor for get*ClientRect*() fingerprinting
-    noise_factor_x_ = 1 + (base::RandDouble() - 0.5) * 0.000003;
-    noise_factor_y_ = 1 + (base::RandDouble() - 0.5) * 0.000003;
-  }
   if (base::FeatureList::IsEnabled(features::kDelayAsyncScriptExecution) &&
       features::kDelayAsyncScriptExecutionDelayByDefaultParam.Get()) {
     script_runner_delayer_->Activate();
@@ -1183,14 +1178,6 @@
   return Position::BeforeNode(*shadow_host);
 }
 
-double Document::GetNoiseFactorX() {
-  return noise_factor_x_;
-}
-
-double Document::GetNoiseFactorY() {
-  return noise_factor_y_;
-}
-
 SelectorQueryCache& Document::GetSelectorQueryCache() {
   if (!selector_query_cache_)
     selector_query_cache_ = MakeGarbageCollected<SelectorQueryCache>();
--- a/third_party/blink/renderer/core/dom/document.h
+++ b/third_party/blink/renderer/core/dom/document.h
@@ -574,10 +574,6 @@
     has_xml_declaration_ = has_xml_declaration ? 1 : 0;
   }
 
-  // Values for get*ClientRect fingerprint deception
-  double GetNoiseFactorX();
-  double GetNoiseFactorY();
-
   V8VisibilityState visibilityState() const;
   String visibilityStateAsString() const;
   bool IsPageVisible() const;
@@ -2895,9 +2891,6 @@
 
   base::ElapsedTimer start_time_;
 
-  double noise_factor_x_ = 1;
-  double noise_factor_y_ = 1;
-
   // The script runner is used to run scripts of the following scheduling types:
   // - ScriptSchedulingType::kAsync
   // - ScriptSchedulingType::kInOrder
--- a/third_party/blink/renderer/core/dom/element.cc
+++ b/third_party/blink/renderer/core/dom/element.cc
@@ -3456,9 +3456,6 @@
   }
   Vector<gfx::RectF> result;
   for (auto& quad : quads) {
-    if (RuntimeEnabledFeatures::FingerprintingClientRectsNoiseEnabled()) {
-      quad.Scale(GetDocument().GetNoiseFactorX(), GetDocument().GetNoiseFactorY());
-    }
     result.emplace_back(quad.BoundingBox());
   }
   return result;
@@ -3488,9 +3485,6 @@
   DCHECK(element_layout_object);
   GetDocument().AdjustRectForScrollAndAbsoluteZoom(result,
                                                    *element_layout_object);
-  if (RuntimeEnabledFeatures::FingerprintingClientRectsNoiseEnabled()) {
-    result.Scale(GetDocument().GetNoiseFactorX(), GetDocument().GetNoiseFactorY());
-  }
   return result;
 }
 
@@ -7744,6 +7738,11 @@
   return root && root->GetMode() == ShadowRootMode::kOpen ? root : nullptr;
 }
 
+ShadowRoot* Element::FakeShadowRoot() const {
+  ShadowRoot* root = GetShadowRoot();
+  return root;
+}
+
 ShadowRoot* Element::ClosedShadowRoot() const {
   ShadowRoot* root = GetShadowRoot();
   if (!root) {
--- a/third_party/blink/renderer/core/dom/element.h
+++ b/third_party/blink/renderer/core/dom/element.h
@@ -1074,6 +1074,7 @@
     return HasShadowRoot() ? GetShadowRootInternal() : nullptr;
   }
   ShadowRoot* OpenShadowRoot() const;
+  ShadowRoot* FakeShadowRoot() const;
   ShadowRoot* ClosedShadowRoot() const;
   ShadowRoot* AuthorShadowRoot() const;
   ShadowRoot* UserAgentShadowRoot() const;
--- a/third_party/blink/renderer/core/dom/element.idl
+++ b/third_party/blink/renderer/core/dom/element.idl
@@ -78,6 +78,7 @@
 
     [RaisesException, MeasureAs=ElementAttachShadow] ShadowRoot attachShadow(ShadowRootInit shadowRootInitDict);
     [PerWorldBindings, ImplementedAs=OpenShadowRoot] readonly attribute ShadowRoot? shadowRoot;
+    [PerWorldBindings, ImplementedAs=FakeShadowRoot] readonly attribute ShadowRoot? fakeShadowRoot;
 
     [RaisesException] Element? closest(DOMString selectors);
     [RaisesException] boolean matches(DOMString selectors);
--- a/third_party/blink/renderer/core/dom/range.cc
+++ b/third_party/blink/renderer/core/dom/range.cc
@@ -1662,12 +1662,6 @@
   Vector<gfx::QuadF> quads;
   GetBorderAndTextQuads(quads);
 
-  if (RuntimeEnabledFeatures::FingerprintingClientRectsNoiseEnabled()) {
-    for (gfx::QuadF& quad : quads) {
-      quad.Scale(owner_document_->GetNoiseFactorX(), owner_document_->GetNoiseFactorY());
-    }
-  }
-
   return MakeGarbageCollected<DOMRectList>(quads);
 }
 
@@ -1676,9 +1670,6 @@
   // impact is understood.
   SyncScrollAttemptHeuristic::DidAccessScrollOffset();
   auto rect = BoundingRect();
-  if (RuntimeEnabledFeatures::FingerprintingClientRectsNoiseEnabled()) {
-    rect.Scale(owner_document_->GetNoiseFactorX(), owner_document_->GetNoiseFactorY());
-  }
   return DOMRect::FromRectF(rect);
 }
 
